### DISCUSSION TITLE
Failed to update filter - 107.33

### DISCUSSION THREAD

**Discussion by jwestyp** [AdguardTeam/AdGuardHome#6015] [General]
Every minute my Adguard server repeats the same behavior and then after some time, fails with a context deadline exceeded error. Logs below:

2023/07/14 05:21:01.011251 [info] dnsforward: starting reconfiguring server
2023/07/14 05:21:01.011260 [info] dnsproxy: stopping dns proxy server
2023/07/14 05:21:01.011538 [info] dnsproxy: stopped dns proxy server
2023/07/14 05:21:01.111844 [info] dnsproxy: cache: enabled, size 4096 b
2023/07/14 05:21:01.111856 [info] dnsproxy: max goroutines is set to 300
2023/07/14 05:21:01.111860 [info] dnsproxy: fastest ip is enabled
2023/07/14 05:21:01.112037 [info] dnsproxy: starting dns proxy server
2023/07/14 05:21:01.112041 [info] Ratelimit is enabled and set to 40 rps
2023/07/14 05:21:01.112043 [info] The server is configured to refuse ANY requests
2023/07/14 05:21:01.112045 [info] dnsproxy: cache: enabled, size 4194304 b
2023/07/14 05:21:01.112047 [info] dnsproxy: max goroutines is set to 300
2023/07/14 05:21:01.112066 [info] dnsproxy: fastest ip is enabled
2023/07/14 05:21:01.112077 [info] dnsproxy: creating udp server socket 0.0.0.0:53
2023/07/14 05:21:01.112129 [info] dnsproxy: listening to udp://[::]:53
2023/07/14 05:21:01.112135 [info] dnsproxy: creating tcp server socket 0.0.0.0:53
2023/07/14 05:21:01.112159 [info] dnsproxy: listening to tcp://[::]:53
2023/07/14 05:21:01.112171 [info] dnsforward: finished reconfiguring server
2023/07/14 05:21:01.112195 [info] dnsproxy: entering udp listener loop on [::]:53
2023/07/14 05:21:01.112253 [info] dnsproxy: entering tcp listener loop on [::]:53
2023/07/14 13:14:50.129658 [info] requesting filter from https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt, skip: Get "https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2023/07/14 13:14:50.129725 [info] Failed to update filter https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt: Get "https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

2023/07/14 14:19:50.130775 [info] requesting filter from https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt, skip: Get "https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2023/07/14 14:19:50.130845 [info] Failed to update filter https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt: Get "https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

2023/07/14 15:24:50.131523 [info] requesting filter from https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt, skip: Get "https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2023/07/14 15:24:50.131608 [info] Failed to update filter https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt: Get "https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

The error will repeat until the docker container is restarted. Since I am using Adguard internally as my DNS, this failure is hard to pin down. Is this caused by adguard dying and not being able to resolve for itself? What are the best ways to determine the root cause so my network doesn't die regularly?

**Comment by Freebase394** [comment]
Could you share your configuration ?
Do you have all ports open ?
